// DAL/commentsFetch.js
import axios from 'axios'
import { getComments, addComment, updateComment, deleteComment } from 'redux/actions/comments'

const token = localStorage.getItem('token')

export const fetchComments = (id) => dispatch => {
   axios({
      method: 'GET',
      url: `https://test.flcd.ru/api/post/${id}/comments`,
      headers: {
         'accept': '*/*',
         'Authorization': `Bearer ${token}`,
         'Content-type': 'application/json',
     }
   })
   .then(res => {
      dispatch(getComments(res.data))
      console.log(res.status)
   })
   .catch(err => {
      console.log(err)
   })
}

export const fetchSetComments = (text, postId) => dispatch => {
   axios({
      method: 'POST',
      url: 'https://test.flcd.ru/api/comment',
      data: {
         text: text,
         post_id: postId,
      },
      headers: {
         'accept': '*/*',
         'Authorization': `Bearer ${token}`,
         'Content-type': 'application/json',
     }
   })
   .then(res => {
      dispatch(addComment(res.data))
      console.log('Комент отправлен', res.status, res.data)
   })
   .catch(err => {
      console.log(err)
   })
}

export const fetchEditComment = (id, text) => dispatch => {
   axios({
      method: 'PATCH',
      url: `https://test.flcd.ru/api/comment/${id}`,
      data: {
         text: text
      },
      headers: {
         'accept': '*/*',
         'Authorization': `Bearer ${token}`,
         'Content-type': 'application/json',
      }
   })
   .then(res => {
      dispatch(updateComment(id, text))
      console.log(res.status)
   })
   .catch(err => {
      console.log(err)
   })
}

export const fetchDeleteComment = (id) => dispatch => {
   axios({
      method: 'DELETE',
      url: `https://test.flcd.ru/api/comment/${id}`,
      headers: {
         'accept': '*/*',
         'Authorization': `Bearer ${token}`,
         'Content-type': 'application/json',
      }
   })
   .then(res => {
      dispatch(deleteComment(id))
      console.log(res.status)
   })
   .catch(err => {
      console.log(err)
   })
}
