// DAL/postsFetching.js
import axios from 'axios'
import { 
    startFetchPosts, 
    successFetchPosts, 
    errorFetchPosts,

    createNewPost,
    updatePost,
    deletePost
} from 'redux/actions/posts'

export const getPosts = () => dispatch => {
    dispatch(startFetchPosts())
    axios({
        method: 'GET',
        url: 'https://test.flcd.ru/api/post'
    })
        .then(res => {
            dispatch(successFetchPosts(res.data))
            console.log('Data was arrived', res.data)
        })

        .catch(err => {
            console.log(err)
            dispatch(errorFetchPosts(err))
        })
}

export const fetchNewPost = (text) => dispatch => {
    axios({
        method: 'POST',
        url: 'https://test.flcd.ru/api/post',
        data: {
            text,
        }
    })
    .then(res => {
        dispatch(createNewPost(res.data.text))
        console.log('New Blog Is Arrived')
    })
    .catch(err => {
        console.log(err) 
    })
}

export const fetchUpdatePost = (text, id) => dispatch => {
    axios({
        method: 'PATCH',
        url: `https://test.flcd.ru/api/${id}`,
        data: {
            text,
        }
    })
    .then((res) => {
        dispatch(updatePost(id, text))
        console.log(`Data post №${id} was update.`)
    })
} 

export const fetchDeletePost = (id) => dispatch => {
    axios({
        method: 'DELETE',
        url: `https://test.flcd.ru/api/${id}`,
        data: {
            id,
        }
    })
    .then(() => {
        dispatch(deletePost(id))
        console.log(`Post №${id} was deleted.`)
    })
}
