// redux/actions/posts.js
import {
   START_FETCH_POSTS,
   SUCCESS_FETCH_POSTS,
   ERROR_FETCH_POSTS,

   CREATE_NEW_POST,
   READ_ONE_POST,
   CLEAR_ONE_POST,
   PATCH_POST,
   DELETE_POST
} from 'redux/actions/constants'

export function startFetchPosts() {
   return {
      type: START_FETCH_POSTS
   }
}

export function successFetchPosts(posts) {
   return {
      type: SUCCESS_FETCH_POSTS,
      payload: {
         posts
      }
   }
}

export function errorFetchPosts(error) {
   return {
      type: ERROR_FETCH_POSTS,
      payload: {
         error
      }
   }
}

export function createNewPost(data) {
   return {
      type: CREATE_NEW_POST,
      payload: data
   }
}

export function readOnePost(data) {
   return {
      type: READ_ONE_POST,
      payload: data
   }
}

export function clearOnePost() {
   return {
      type: CLEAR_ONE_POST,
   }
}

export function updatePost(id, text) {
   
   return {
      type: PATCH_POST,
      payload: {
         id, text
      }
   }
}

export function deletePost(id) {
   return {
      type: DELETE_POST,
      payload: {
         id
      }
   }
}
